ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp

```diff
@@ -57,19 +67,38 @@ void show_error(string errortext, const bool fatal)
   #ifdef DEBUG_MODE
     errortext += enigma::debug_scope::GetErrors();
   #else
-  errortext = "Error in some event or another for some object: \r\n" + errortext;
+    errortext = "Error in some event or another for some object: \r\n" + errortext;
   #endif
 
-  if (MessageBox(NULL,errortext.c_str(),"Error",MB_ABORTRETRYIGNORE | MB_ICONERROR)==IDABORT)
-    exit(0);
+  string strStr = errortext;
+  string strWindowCaption = "Error";
 
-  if (fatal)
-    printf("FATAL ERROR: %s\n",errortext.c_str()),
-    exit(0);
+  if (strStr != "")
+    strStr = strStr + "\n\n";
+
+  if (fatal == 0)
+    strStr = strStr + "Do you want to abort the application?";
   else
-    printf("ERROR: %s\n",errortext.c_str());
+    strStr = strStr + "Click 'OK' to abort the application.";
```

Comment:
The idea behind the `fatal` parameter is that if it's `true` it is _impossible_ for ENIGMA to recover. Everything else should be considered non-fatal.